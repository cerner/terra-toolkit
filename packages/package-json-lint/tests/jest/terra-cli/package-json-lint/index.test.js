jest.mock('../../../../src/lint');

const yargs = require('yargs');
const lint = require('../../../../src/lint');
const PackageJsonLintCommand = require('../../../../src/terra-cli/package-json-lint');

describe('package-json-lint', () => {
  it('declares the package-json-lint terra-cli command with proper top level help', async () => {
    const parser = yargs.command(PackageJsonLintCommand).scriptName('terra').help();
    const helpOutput = await new Promise((resolve) => {
      parser.parse('--help', (err, argv, output) => {
        resolve(output);
      });
    });
    expect(helpOutput).toMatchSnapshot();
  });

  it('declares the package-json-lint terra-cli command with proper command level help', async () => {
    const parser = yargs.command(PackageJsonLintCommand).scriptName('terra').help();
    const helpOutput = await new Promise((resolve) => {
      parser.parse('package-json-lint --help', (err, argv, output) => {
        resolve(output);
      });
    });
    expect(helpOutput).toMatchSnapshot();
  });

  it('uses the appropriate handler when calling package-json-lint', async () => {
    await PackageJsonLintCommand.handler();
    expect(lint).toHaveBeenCalled();
  });
});
