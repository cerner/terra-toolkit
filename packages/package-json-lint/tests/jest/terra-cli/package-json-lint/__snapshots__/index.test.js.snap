// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`package-json-lint declares the package-json-lint terra-cli command with proper command level help 1`] = `
"terra package-json-lint

Lint the package.json file(s) in the current package

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]"
`;

exports[`package-json-lint declares the package-json-lint terra-cli command with proper top level help 1`] = `
"terra [command]

Commands:
  terra package-json-lint  Lint the package.json file(s) in the current package

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]"
`;
